test/parallel/test-whatwg-url-from.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const { URL } = require('url');
+
+function t(expectedUrl, actualConfig) {
+  const url = URL.from(actualConfig);
+  assert.strictEqual(String(url), expectedUrl);
+}
+
+t(':', {});
+t(':', []);
+t(':', undefined);
```

Comment:
Most of these should be type errors. `typeof input === 'object' && input !== null`.

Reply:
Yeah, as mentioned above I will restrict everything except `{}`.